crates/core/src/rlp/decode.rs

```diff
@@ -49,174 +67,188 @@ impl RLPDecode for u8 {
 }
 
 impl RLPDecode for u16 {
-    fn decode(rlp: &[u8]) -> Result<Self, RLPDecodeError> {
-        let (bytes, _) = decode_bytes(rlp)?;
+    fn decode_unfinished(rlp: &[u8]) -> Result<(Self, &[u8]), RLPDecodeError> {
+        let (bytes, rest) = decode_bytes(rlp)?;
```

Comment:
We need to check inside `decode_bytes` that the RLP item is not a list. If not, `0xc101` could be wrongly interpreted as `1` instead of `[1]`

Reply:
I'll add a comment to warn users about that. Since Vec<T> are interpreted only as lists.

Reply:
comments added